### ISSUE TITLE
After upgrade to Jekyll 3.0.0, Conflict with jekyll-compass occuring

### ISSUE THREAD

**Issue by fxrdesign** [mscharley/jekyll-compass#33]
[Originally posted here with direction to repost issue](https://github.com/jekyll/jekyll/issues/4073)

I updated my Jekyll gem this morning with no issues during install. However when I attempted to `jekyll serve --watch` I received the following error: (note that I had the jekyll-compass plugin previously installed with no issues generating my site)

```
Configuration file: /path/to/my/jekyll/project/_config.yml
/Library/Ruby/Site/2.0.0/rubygems/specification.rb:2112:in `raise_if_conflicts': Unable to activate jekyll-compass-1.2.0, because jekyll-3.0.0 conflicts with jekyll (~> 2.0) (Gem::ConflictError)
	from /Library/Ruby/Site/2.0.0/rubygems/specification.rb:1280:in `activate'
	from /Library/Ruby/Site/2.0.0/rubygems.rb:198:in `rescue in try_activate'
	from /Library/Ruby/Site/2.0.0/rubygems.rb:195:in `try_activate'
	from /Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_require.rb:126:in `rescue in require'
	from /Library/Ruby/Site/2.0.0/rubygems/core_ext/kernel_require.rb:39:in `require'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/plugin_manager.rb:30:in `block in require_gems'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/plugin_manager.rb:27:in `each'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/plugin_manager.rb:27:in `require_gems'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/plugin_manager.rb:19:in `conscientious_require'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/site.rb:97:in `setup'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/site.rb:49:in `initialize'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/commands/build.rb:30:in `new'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/commands/build.rb:30:in `process'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/lib/jekyll/commands/serve.rb:26:in `block (2 levels) in init_with_program'
	from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
	from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
	from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
	from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
	from /Library/Ruby/Gems/2.0.0/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-3.0.0/bin/jekyll:17:in `<top (required)>'
	from /usr/bin/jekyll:23:in `load'
	from /usr/bin/jekyll:23:in `<main>'
```
I attempted to reinstall jekyll-compass, but this had no effect on the error I receive.

Currently using Option 2 for [plugin installation](http://jekyllrb.com/docs/plugins/) and it looks like this in my _config.yml file:

`gems: [jekyll-compass, jekyll-redirect-from]`

**Comment by envygeeks** [comment]
Now that this ticket is here I can help guide you: The reason this doesn't bubble during install is because you are using Global install through RubyGem itself rather than through something like Bundler which will surface the error while doing depcheck.

When you globally install it installs Jekyll 2.5 next to Jekyll 3.0 that you requested, but you get the latest Jekyll because that's what RubyGem selected for you (because it was the one that was directly installed.)  So what happens here is you install jekyll-compass and jekyll and then jekyll-compass installs jekyll 2.0 but you installed 3.0 but you have us (Jekyll 3.0) load jekyll-compass and finally Gem surfaces the error because it's at that point that Gem finally does a depcheck and enforces the constraints that jekyll-compass wanted.

In the mean time (having done a quick run of the source without paying much attention -- because there are no specs) this uses a generator, so I'm going to assume it's safe for you to fork this gem, run `gem build jekyll-compass.gemspec` gem install *.gem and then go on about your way after editing the gemspec to use Jekyll 3.0.

**Comment by fxrdesign** [comment]
Thank you for the help @envygeeks unfortunately I am very much a noob at working with gems/updating code on git. So far I have been able to fork the gem on github, and update the jekyll-compass.gemspec file, but now I am not clear on how to implement that change to my local version of jekyll-compass. Tried `gem build jekyll-compass.gemspec` and got the error `ERROR:  Gemspec file not found: jekyll-compass.gemspec`. 

Looks like I will need a bit more hand-holding to get this working - if you have the time.
